go/enclave/enclave.go

```diff
@@ -485,51 +494,54 @@ func (e *enclaveImpl) GetTransactionCount(encryptedParams common.EncryptedParams
 		//  conditions we allow it to return zero while head state is uninitialized
 		s, err := e.storage.CreateStateDB(*l2Head.Hash())
 		if err != nil {
-			err = fmt.Errorf("could not create stateDB. Cause: %w", err)
-			return responses.AsPlaintextError(err)
+			if errors.Is(err, syserr.InternalError{}) {
+				return nil, responses.ToInternalError(err)
+			}
+			return responses.AsPlaintextError(fmt.Errorf("could not create stateDB. Cause: %w", err)), nil
```

Comment:
this error can't be a user error under any circumstance.
if the statedb fails , it should only return a syserror